File: dart/bob.dart
class Bob {
  String response(String input) {
    input = input.trim();
    bool is_question = input.endsWith('?');
    bool is_yelling =
        RegExp(r'[A-Za-z]').hasMatch(input) && input.toUpperCase() == input;
    bool is_yelling_question = is_question && is_yelling;
    bool is_silence = input == '';

    if (is_yelling_question) {
      return "Calm down, I know what I'm doing!";
    }

    if (is_question) {
      return "Sure.";
    }

    if (is_yelling) {
      return "Whoa, chill out!";
    }

    if (is_silence) {
      return "Fine. Be that way!";
    }

    return "Whatever.";
  }
}

File: cs/Bob.cs
using System;
using System.Text.RegularExpressions;

public static class Bob
{
    public static string Response(string statement)
    {
        statement = statement.Trim();
        if (statement == "")
            return "Fine. Be that way!";

        bool hasLetter = Regex.IsMatch(statement, @"[a-zA-Z]");
        bool isQuestion = statement.EndsWith("?");
        bool isYelling = hasLetter && statement.ToUpper() == statement;
        if (isQuestion && isYelling)
            return "Calm down, I know what I'm doing!";
        if (isQuestion)
            return "Sure.";
        if (isYelling)
            return "Whoa, chill out!";
        return "Whatever.";
    }
}
